eng/pipelines/aggregate-reports.yml

```diff
@@ -168,6 +168,26 @@ stages:
                 -WorkingDirectory '$(Build.ArtifactStagingDirectory)'
                 -NupkgFilesDestination 'nupkgFiles'
 
+          - task: AzureCLI@2
+            displayName: Azure CLI Login
+            inputs:
+              azureSubscription: azure-sdk-apiscan
+              scriptType: pscore
+              scriptLocation: inlineScript
+              addSpnToEnvironment: true
+              inlineScript: |
+                az --version
+                az account show
+                Write-Host "##vso[task.setvariable variable=ARM_CLIENT_ID;issecret=true]$($env:servicePrincipalId)"
+                Write-Host "##vso[task.setvariable variable=ARM_TENANT_ID;issecret=true]$($env:tenantId)"
+                Write-Host "##vso[task.setvariable variable=ARM_OIDC_TOKEN;issecret=true]$($env:idToken)"
+
+          - pwsh: |
+                az login --service-principal -u $(ARM_CLIENT_ID) --tenant $(ARM_TENANT_ID) --allow-no-subscriptions --federated-token $(ARM_OIDC_TOKEN)
```

Comment:
@benbp can you double check this work and let me know if this is too hacky? At the root I need to have a login credential to pass to the apiscan task. The CLI nor PS stay logged in so I'm exporting the token to login and using it here.

Reply:
I think this is fine, assuming the task doesn't take longer than the token expiry. I'm pretty much using this same az login approach in other places where to bridge the workload identity support gap.